skyrim-platform/src/platform_se/skyrim_platform/GameEventSinks.cpp

```diff
@@ -665,3 +732,1297 @@ RE::BSEventNotifyControl GameEventSinks::ProcessEvent(
+RE::BSEventNotifyControl GameEventSinks::ProcessEvent(
+  const SKSE::CrosshairRefEvent* event,
+  RE::BSTEventSource<SKSE::CrosshairRefEvent>* eventSource)
+{
+  if (event == nullptr) {
+    return RE::BSEventNotifyControl::kContinue;
+  }
+
+  auto ref = event->crosshairRef.get();
+  auto refId = ref ? ref->formID : 0;
+
+  SkyrimPlatform::GetSingleton().AddUpdateTask([ref, refId] {
+    auto obj = JsValue::Object();
+
+    auto refLocal = RE::TESForm::LookupByID(refId);
+
+    if (refId == 0) {
+      obj.SetProperty("reference", JsValue::Null());
+    } else {
+      if (refLocal == nullptr || refLocal != ref) {
+        return;
+      }
+      obj.SetProperty("reference", CreateObject("ObjectReference", refLocal));
+    }
+
+    EventsApi::SendEvent("crosshair", { JsValue::Undefined(), obj });
+  });
+
+  return RE::BSEventNotifyControl::kContinue;
+}
+
+RE::BSEventNotifyControl GameEventSinks::ProcessEvent(
+  const SKSE::NiNodeUpdateEvent* event,
+  RE::BSTEventSource<SKSE::NiNodeUpdateEvent>* eventSource)
+{
+  if (event == nullptr) {
+    return RE::BSEventNotifyControl::kContinue;
+  }
+
+  auto ref = event->reference;
+  auto refId = ref ? ref->formID : 0;
+
+  SkyrimPlatform::GetSingleton().AddUpdateTask([ref, refId] {
+    auto obj = JsValue::Object();
+
+    auto refLocal = RE::TESForm::LookupByID(refId);
+
+    if (refId == 0) {
+      obj.SetProperty("reference", JsValue::Null());
+    } else {
+      if (refLocal == nullptr || refLocal != ref) {
+        return;
+      }
+      obj.SetProperty("reference", CreateObject("ObjectReference", refLocal));
+    }
+
+    EventsApi::SendEvent("niNodeUpdate", { JsValue::Undefined(), obj });
+  });
+
+  return RE::BSEventNotifyControl::kContinue;
+}
+
+RE::BSEventNotifyControl GameEventSinks::ProcessEvent(
+  const SKSE::ModCallbackEvent* event,
+  RE::BSTEventSource<SKSE::ModCallbackEvent>* eventSource)
+{
+  if (event == nullptr) {
+    return RE::BSEventNotifyControl::kContinue;
+  }
+
+  auto eventName = event->eventName;
+  auto numArg = event->numArg;
+  auto strArg = event->strArg;
+  auto sender = event->sender;
+  auto senderId = sender ? sender->formID : 0;
+
+  SkyrimPlatform::GetSingleton().AddUpdateTask(
+    [eventName, numArg, strArg, sender, senderId] {
+      auto obj = JsValue::Object();
+
+      auto senderLocal = RE::TESForm::LookupByID(senderId);
+
+      if (senderLocal == nullptr || senderLocal != sender) {
+        return;
+      }
+
+      obj.SetProperty("sender", CreateObject("Form", senderLocal));
+      obj.SetProperty("eventName",
+                      JsValue::String(static_cast<std::string>(eventName)));
+      obj.SetProperty("strArg",
+                      JsValue::String(static_cast<std::string>(strArg)));
+      obj.SetProperty("numArg", JsValue::Double(numArg));
+
+      EventsApi::SendEvent("modEvent", { JsValue::Undefined(), obj });
+    });
+
+  return RE::BSEventNotifyControl::kContinue;
+}
+
+RE::BSEventNotifyControl GameEventSinks::ProcessEvent(
```

Comment:
These two events look strange. I think it needs more in-game testing and can be landed in a separate PR.
Please remove from this PR for now.